pkg/api/v20191231preview/openshiftcluster_convert.go

```diff
@@ -117,42 +117,28 @@ func (c *openShiftClusterConverter) ToInternal(_oc interface{}, out *api.OpenShi
 	out.Properties.NetworkProfile.ServiceCIDR = oc.Properties.NetworkProfile.ServiceCIDR
 	out.Properties.MasterProfile.VMSize = api.VMSize(oc.Properties.MasterProfile.VMSize)
 	out.Properties.MasterProfile.SubnetID = oc.Properties.MasterProfile.SubnetID
-	for _, p := range oc.Properties.WorkerProfiles {
-		var outp *api.WorkerProfile
-		for i, pp := range out.Properties.WorkerProfiles {
-			if pp.Name == p.Name {
```

Comment:
I think, we don't need custom marshaling: Go's standard marshaling will work well, given that we agreed that patch should not do magic merging server side.